src/components/DefineCompany/OfficeAddresses.vue

```diff
@@ -0,0 +1,518 @@
+
+  // Entity Enum
+  readonly EntityTypes: {} = EntityTypes;
+
+  /**
+   * Lifecycle callback to set up the component when it is mounted.
+   */
+  private mounted (): void {
+    if (this.addresses) {
+      this.deliveryAddress = this.addresses.registeredOffice.deliveryAddress
+      this.mailingAddress = this.addresses.registeredOffice.mailingAddress
+      this.inheritMailingAddress = this.isSame(
+        this.addresses.registeredOffice.deliveryAddress,
+        this.addresses.registeredOffice.mailingAddress)
+
+      if (this.entityFilter(EntityTypes.BCOMP)) {
+        this.recDeliveryAddress = this.addresses.recordsOffice!.deliveryAddress
+        this.recMailingAddress = this.addresses.recordsOffice!.mailingAddress
+        this.inheritRegisteredAddress = (
+          this.isSame(this.addresses.registeredOffice.deliveryAddress, this.addresses.recordsOffice!.deliveryAddress) &&
+          this.isSame(this.addresses.registeredOffice.mailingAddress, this.addresses.recordsOffice!.mailingAddress)
+        )
+        this.inheritRecMailingAddress = this.isSame(
+          this.addresses.recordsOffice!.deliveryAddress,
+          this.addresses.recordsOffice!.mailingAddress)
+      }
+    }
+    this.emitValid()
+  }
+
+  // Computed values.
+  /**
+   * Computed value of whether or not the address form is valid.
+   *
+   * @returns a boolean that is true if the data on the form is valid, or false otherwise.
+   */
+  private get formValid (): boolean {
+    return (
+      this.deliveryAddressValid &&
+      (this.inheritMailingAddress || this.mailingAddressValid) &&
+      this.recDeliveryAddressValid && (this.inheritRecMailingAddress || this.recMailingAddressValid)
+    )
+  }
+
+  private isEmptyAddress (address: AddressIF) : boolean {
+    return isEmpty(address)
+  }
+
+  // Event Handlers.
+  /**
+   * Event handler that sets the Registered Office Mailing Address to
+   * Registered Office Delivery Address.
+   */
+  private setDeliveryAddressToMailingAddress () : void {
+    if (this.inheritMailingAddress) {
+      this.deliveryAddress = { ...this.mailingAddress }
+      this.emitAddresses()
+    }
+  }
+
+  // Event Handler that sets the record office to registered office
+  private setRecordOfficeToRegisteredOffice () : void {
+    if (this.inheritRegisteredAddress) {
+      this.recDeliveryAddress = { ...this.deliveryAddress }
+      this.recMailingAddress = { ...this.mailingAddress }
+      this.emitAddresses()
+    } else {
+      this.inheritRecMailingAddress = this.inheritMailingAddress
+    }
+  }
+
+  private setRecordDeliveryAddressToMailingAddress () : void {
+    if (this.inheritRecMailingAddress) {
+      this.recDeliveryAddress = { ...this.recMailingAddress }
+      this.emitAddresses()
+    }
+  }
+
+  private updateAddress (addressChangedEntity:string, baseAddress: AddressIF, newAddress: AddressIF): void {
+    Object.assign(baseAddress, newAddress)
+    switch (addressChangedEntity) {
+      case 'mailingAddress':
```

Comment:
Please use constant. Also in calls to this method. Also in cases below.

Also in updateAddressValid() below.